test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceTests.cs

```diff
@@ -393,6 +403,174 @@ public async void SchemaComparePublishChangesDatabaseToDatabase()
+        {
+            SqlTestDb sourceDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, SourceScript, "SchemaCompareSource");
+            SqlTestDb targetDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, TargetScript, "SchemaCompareTarget");
+
+            try
+            {
+                string sourceDacpac = SchemaCompareTestUtils.CreateDacpac(sourceDb);
+                string targetDacpac = SchemaCompareTestUtils.CreateDacpac(targetDb);
+                string filePath = SchemaCompareTestUtils.CreateScmpPath();
+
+                SchemaCompareEndpointInfo sourceInfo = new SchemaCompareEndpointInfo();
+                SchemaCompareEndpointInfo targetInfo = new SchemaCompareEndpointInfo();
+
+                sourceInfo.EndpointType = SchemaCompareEndpointType.Dacpac;
+                sourceInfo.PackageFilePath = sourceDacpac;
+                targetInfo.EndpointType = SchemaCompareEndpointType.Dacpac;
+                targetInfo.PackageFilePath = targetDacpac;
+
+                CreateAndValidateScmpFile(sourceInfo, targetInfo, false);
+
+            }
+            finally
+            {
+                sourceDb.Cleanup();
+                targetDb.Cleanup();
+            }
+        }
+
+        /// <summary>
+        /// Verify the schema compare Service Calls ends to end
+        /// </summary>
+        [Fact]
+        public async Task VerifySchemaCompareServiceCalls()
+        {
+            string operationId = null;
+            DiffEntry diffEntry = null;
+            var connectionObject = SchemaCompareTestUtils.GetLiveAutoCompleteTestObjects();
+
+            SqlTestDb sourceDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, SourceScript, "SchemaCompareSource");
+            SqlTestDb targetDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, TargetScript, "SchemaCompareTarget");
+
+            try
+            {
+                SchemaCompareEndpointInfo sourceInfo = new SchemaCompareEndpointInfo();
+                SchemaCompareEndpointInfo targetInfo = new SchemaCompareEndpointInfo();
+
+                sourceInfo.EndpointType = SchemaCompareEndpointType.Database;
+                sourceInfo.DatabaseName = sourceDb.DatabaseName;
+                sourceInfo.OwnerUri = connectionObject.ConnectionInfo.OwnerUri;
+                targetInfo.EndpointType = SchemaCompareEndpointType.Database;
+                targetInfo.DatabaseName = targetDb.DatabaseName;
+                targetInfo.OwnerUri = connectionObject.ConnectionInfo.OwnerUri;
+
+                // Schema compare service call
+                var schemaCompareRequestContext = new Mock<RequestContext<SchemaCompareResult>>();
+                schemaCompareRequestContext.Setup((RequestContext<SchemaCompareResult> x) => x.SendResult(It.Is<SchemaCompareResult>((diffResult) =>
+                ValidateScResult(diffResult, ref diffEntry, ref operationId)))).Returns(Task.FromResult(new object()));
```

Comment:
should these be out instead of ref since they get set for the first time in ValidateScResult()?

Reply:
will do